kernel/cemitcodes.ml

```diff
@@ -369,10 +369,17 @@ let to_memory (init_code, fun_code, fv) =
   init();
   emit init_code;
   emit fun_code;
-  let code = String.create !out_position in
-  String.unsafe_blit !out_buffer 0 code 0 !out_position;
   (** Later uses of this string are all purely functional *)
-  let code = CString.hcons code in
+  let code = Bytes.init !out_position (fun i ->
+      Bytes.unsafe_get !out_buffer i
+    ) in
+  (* XXX: We lose the hconsing optimization wrt the non-bytes version.
+
+     This really depends on whether we want emitcodes to be `string` or `bytes`.
+
+     The current choice is `bytes`, but `string` can be done using a
+     more careful patching. *)
```

Comment:
The problem here is that we would introduce an extra copy in `patch_int`, right? That may turn out to be more costly than losing the hash-consing.